plugins/com.google.cloud.tools.eclipse.appengine.ui/src/com/google/cloud/tools/eclipse/appengine/ui/LibrarySelectorGroup.java

```diff
@@ -73,19 +86,62 @@ private void createContents(Composite parentContainer) {
     Group apiGroup = new Group(parentContainer, SWT.NONE);
     apiGroup.setText(Messages.getString("appengine.libraries.group"));
 
+    IProject project = getSelectedProject();
     for (Library library : availableLibraries.values()) {
-      Button libraryButton = new Button(apiGroup, SWT.CHECK);
-      libraryButton.setText(getLibraryName(library));
-      if (library.getToolTip() != null) {
-        libraryButton.setToolTipText(library.getToolTip());
+      if (enabled(library, project)) {
+        Button libraryButton = new Button(apiGroup, SWT.CHECK);
+        libraryButton.setText(getLibraryName(library));
+        if (library.getToolTip() != null) {
+          libraryButton.setToolTipText(library.getToolTip());
+        }
+        libraryButton.setData(library);
+        libraryButton.addSelectionListener(new ManualSelectionTracker());
+        libraryButtons.put(library, libraryButton);
       }
-      libraryButton.setData(library);
-      libraryButton.addSelectionListener(new ManualSelectionTracker());
-      libraryButtons.put(library, libraryButton);
     }
     GridLayoutFactory.swtDefaults().generateLayout(apiGroup);
   }
 
+  private static boolean enabled(Library library, IProject project) {
+    Expression expression = library.getEnablement();
+    if (expression == null) {
+      return true; 
+    }
+
+    // fill in the default variable with the currently selected project
+    List<IProject> defaultVariable = new ArrayList<>();
+    if (project != null) {
+      defaultVariable.add(project);
+    }
+    IEvaluationContext context = new EvaluationContext(null, defaultVariable);
+    try {
+      EvaluationResult result = expression.evaluate(context);
+      return result == EvaluationResult.TRUE;
+    } catch (CoreException ex) {
+      // default to including the library
+      return true;
+    }
+  }
+
+  private static IProject getSelectedProject() {
+    // todo is this the best way to grab the selection? If so move to a utility package? 
+    IWorkbench workbench = PlatformUI.getWorkbench();
+    IWorkbenchWindow activeWorkbenchWindow = workbench.getActiveWorkbenchWindow();
+    IWorkbenchPage activePage = activeWorkbenchWindow.getActivePage();
+    ISelection selection = activePage.getSelection();
+    if (selection instanceof IStructuredSelection) {
+      IStructuredSelection structuredSelection = (IStructuredSelection) selection;
+      Object element = structuredSelection.getFirstElement();
+      if (element instanceof IResource) {
```

Comment:
This can be simplified further. Take a look at `ProjectFromSelectionHelper.getProject()`.